commons-component-common/src/main/java/org/exoplatform/commons/notification/impl/jpa/email/JPAQueueMessageImpl.java

```
@@ -0,0 +1,272 @@
+
+      mailQueueDAO.create(mailQueueEntity);
+
+    } catch (Exception e) {
+      LOG.error("Failed to save message.", e.getMessage() + message.toJSON(), e);
+    }
+  }
+
+  @Override
+  @ExoTransactional
+  public void send() {
+    final boolean stats = NotificationContextFactory.getInstance().getStatistics().isStatisticsEnabled();
+    try {
+      //
+      load();
+      if (idsRemovingLocal.get() == null) {
+        idsRemovingLocal.set(new HashSet<String>());
+      }
+      //
+      if (messages.size() > 0) {
+        LOG.info(messages.size() + " message(s) will be sent.");
+      }
+
+      for (MessageInfo messageInfo : messages) {
+        if (messageInfo != null && !idsRemovingLocal.get().contains(messageInfo.getId())
+            && sendMessage(messageInfo.makeEmailNotification())) {
+
+          LOG.debug("Message sent to user: " + messageInfo.getTo());
+          //
+          idsRemovingLocal.get().add(messageInfo.getId());
+          if (stats) {
+            NotificationContextFactory.getInstance().getStatisticsCollector().pollQueue(messageInfo.getPluginId());
+          }
+        }
+      }
+    } catch (Exception e) {
+      LOG.warn("Failed to send message.");
+      LOG.debug(e.getMessage(), e);
+    }
+    removeMessageInfo();
+  }
+
+  private void load() {
+    try {
+      for (MailQueueEntity mailQueueEntity : mailQueueDAO.findAll(0, LIMIT)) {
+        messages.add(convertQueueEntityToMessageInfo(mailQueueEntity));
+      }
+    } catch (Exception e) {
+      LOG.error("Failed to load message.", e.getMessage(), e);
+    }
+  }
+
+  @Override
+  public boolean sendMessage(Message message) {
+    if (!sendEmailService.isOn()) {
+      try {
+        //ensure the message is valid
+        if (message.getFrom() == null) {
+          return false;
+        }
+        mailService.sendMessage(message);
+        return true;
+      } catch (Exception e) {
+        LOG.error("Error while sending a message - Cause : " + e.getMessage(), e);
```

The error management will be treated apart in PLF-7389.
But this method shouldn't swallow the exception and should throw it to caller method.